### DISCUSSION TITLE
Reopening - [BUG] New-PnPSite not working with HubSiteID > Unauthorized

### DISCUSSION THREAD

**Discussion by NoralK** [pnp/powershell#3843] [General]
### Reporting an Issue or Missing Feature
Reopening #3594

I'm unable to run New-PnpSite when providing a `-hubSiteID`. When I run it without providing the `-HubSiteID`, it works. I have the SharePoint Admin role active and in the SharePoint admin center I can perform this action.

More information - GCC Tenant - The hub does not require authorization, as the SharePoint Admin I have created and am a admin over all site collections.

### Expected behavior 
Create a site with a hub association during creation.

### Actual behavior 
```
Line |
  32 |  New-PnPSite -Url "https://orgname.sharepoint.com/sites/testTeam" `
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | {"error":{"code":"-2147024891, System.UnauthorizedAccessException","message":"Attempted to perform an unauthorized
     | operation."}}
```

### Steps to reproduce behavior
```
Import-Module -Name PnP.PowerShell
Connect-PnPOnline https://orgname-admin.sharepoint.com/ -Interactive

$hub = Get-PnPHubSite -Identity "https://orgname.sharepoint.com/sites/IT"

New-PnPSite -Url "https://orgname.sharepoint.com/sites/testTeam" `
    -Type TeamSiteWithoutMicrosoft365Group `
    -Title "Test Team Site" `
    -Description "Test Team Site Description" `
    -Owner "user@org.com" `
    -HubSiteId $hub.SiteId #00000000-0000-0000-0000-000000000000 FYI GUID is scrubbed
```
I performed the suggestion by @gautamdsheth to use the GUID to create the site, this did not work.   As mentioned by OP and now me, running `Add-PnPHubSiteAssociation` after the site creation succeeds.

### What is the version of the Cmdlet module you are running?
In PS-ISE - v1.12.0
In vsCode and PowerShell v7.4.1 - v2.3.0

### Which operating system/environment are you running PnP PowerShell on?
- [ X ] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by gautamdsheth** [comment]
@veronicageek - can you also try and reproduce this issue ? I couldn't , so not sure if we can do anything about this. For me, it works everytime.

**Comment by NoralK** [comment]
@gautamdsheth  @veronicageek  - Could it be due to our Tenant being a GCC?

**Comment by gautamdsheth** [comment]
No idea, can you maybe try setting the owner parameter the same as the one you use to connect to and try again ?

**Comment by NoralK** [comment]
My bad, I need to update to original code.  I specified the owner being an email address when in fact it is an Azure AD Group.
```
New-PnPSite -Url "https://orgname.sharepoint.com/sites/testTeam" `
    -Type TeamSiteWithoutMicrosoft365Group `
    -Title "Test Team Site" `
    -Description "Test Team Site Description" `
    -Owner "<<AzureAD Group Display Name>>" `
    -HubSiteId $hub.SiteId #00000000-0000-0000-0000-000000000000 FYI GUID is scrubbed
```
Therefore the `(Get-PnPSite).Owner` is blank.  I looked at the documenation again, `New-PnPSite ... (parms) ... -Owner ... If this parameter is skipped or a group object is provided, the user running New-PnPSite command will be set as a site owner.`, this is not happening, again it is blank.

**Comment by NoralK** [comment]
@gautamdsheth  @veronicageek  any updates?

**Comment by gautamdsheth** [comment]
No updates and we can't repro it either, will move it to discussion forum in case someone in the community has idea.

**Comment by J03rg** [comment]
We are encountering the exact same issue. However, we noticed that changing a specific setting in the SharePoint admin center resolves it immediately. We've tested this multiple times, and it consistently shows the same behavior.

Disabling "Show the options to create a site in SharePoint and create a shared library from OneDrive" and running:
`New-PnPSite -Url "https://orgname.sharepoint.com/sites/test-5" -Type CommunicationSite -Owner "AAD-Group" -Title "Test5" -HubSiteId "af3edcd7-105d-44b9-9fa9-19d8989ac617"`
throws the same error:
`{"error":{"code":"-2147024891, System.UnauthorizedAccessException","message":"Attempted to perform an unauthorized operation."`

Enabling the setting and running the same command works fine. 
Not sure how or why this setting relates to the hub site association, but we tried it multiple times in a row and as soon as we enable the setting, it works immediately. 

Mentioned Setting: https://learn.microsoft.com/en-us/sharepoint/manage-site-creation
"Users can create SharePoint sites" is enabled and it doesn't cause an issue. 
Only the second option, "Show the options to create a site in SharePoint and create a shared library from OneDrive," causes the issue.

**Comment by michhes** [comment]
Same issue here when provisioning a CommunicationSite (TeamSite works fine). Notably running Add-PnPHubSiteAssociatio after creating a communication site functions as expected.

New-PnPSite -Type CommunicationSite -Title 'Test' -Url 'https://mytenant.sharepoint.com/sites/test' -HubSiteId '30c5bf44-b5aa-427f-a21b-ec9366aaa329'
New-PnPSite: {"error":{"code":"-2147024891, System.UnauthorizedAccessException","message":"Attempted to perform an unauthorized operation."}}
